ui/src/app/controllers/views/ControllerList/ControllerListTable/VersionColumn/VersionColumn.tsx

```diff
@@ -0,0 +1,65 @@
+import { Icon, Tooltip } from "@canonical/react-components";
+import { useSelector } from "react-redux";
+
+import DoubleRow from "app/base/components/DoubleRow";
+import controllerSelectors from "app/store/controller/selectors";
+import type { Controller, ControllerMeta } from "app/store/controller/types";
+import type { RootState } from "app/store/root/types";
+
+type Props = {
+  systemId: Controller[ControllerMeta.PK];
+};
+
+export const VersionColumn = ({ systemId }: Props): JSX.Element | null => {
+  const controller = useSelector((state: RootState) =>
+    controllerSelectors.getById(state, systemId)
+  );
+  const versions = controller?.versions;
+  if (!versions) {
+    return null;
+  }
+  let cohortKey = null;
+  if (versions.snap_cohort) {
+    // Format the key into lines of 41 characters so that it can be displayed
+    // nicely in the tooltip.
+    const chunks = versions.snap_cohort.match(/.{1,41}/g) || [];
+    cohortKey = chunks.map((chunk) => chunk.trim()).join(" \n");
+  }
+  const origin = versions.origin || null;
+  const cohortTooltip = cohortKey ? `Cohort key: \n${cohortKey}` : null;
+  const isDeb = versions.install_type === "deb";
+  return (
+    <DoubleRow
+      primary={
+        <span data-testid="version">
+          {versions.current.version ?? (
```

Comment:
We should add `primaryTitle` or a tooltip since with all the extra version/revision data this string can get pretty long

Reply:
Done.